sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/RepositoryHandler.java

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.iot.modelsrepository.implementation;
+
+import com.azure.core.exception.AzureException;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.iot.modelsrepository.DtmiConventions;
+import com.azure.iot.modelsrepository.ModelsDependencyResolution;
+import com.azure.iot.modelsrepository.implementation.models.FetchResult;
+import com.azure.iot.modelsrepository.implementation.models.ModelMetadata;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.util.*;
+
+public final class RepositoryHandler {
+
+    private final URI repositoryUri;
+    private final ModelFetcher modelFetcher;
+    private final ClientLogger logger;
+
+    public RepositoryHandler(URI repositoryUri, ModelsRepositoryAPIImpl protocolLayer, ClientLogger logger) {
+        this.repositoryUri = repositoryUri;
+        this.logger = logger;
+
+        if (this.repositoryUri.getScheme().toLowerCase().startsWith(ModelsRepositoryConstants.FILE)) {
+            this.modelFetcher = new FileModelFetcher(this.logger);
+        } else {
+            this.modelFetcher = new HttpModelFetcher(protocolLayer, this.logger);
+        }
+    }
+
+    public Mono<Map<String, String>> processAsync(String dtmi, ModelsDependencyResolution resolutionOptions, Context context) {
+        return processAsync(Collections.singletonList(dtmi), resolutionOptions, context);
+    }
+
+    // This doesn't work as it throws it into an infinite loop
```

Comment:
Is this a TODO?

Reply:
no longer valid, will remove :) good catch